htdocs/js/ui_utils.js

```diff
@@ -182,6 +183,41 @@ ui_utils.install_common_ace_key_bindings = function(widget, get_language) {
                         shell.scroll_to_end();
                     });
             }
+        },
+
+        {
+            name: 'execute-cursor-at-beginning',
+            bindKey: {
+          
+                mac: 'Ctrl-A',
+                sender: 'editor'
+            },
+            exec: function(widget, args, request) {
+                if (widget.getOption("readOnly"))
+                    return;
+                //row of the cursor on current line
+                var row = widget.getCursorPosition().row;
+                //move to the beginning of that line
+                widget.navigateTo(row, 0);
+                //make sure it appears at beginning of text
+                widget.navigateLineStart();
+            }
+        } ,
+        {
+            name: 'execute-cursor-at-end',
+            bindKey: {
+       
+                mac: 'Ctrl-E',
+                sender: 'editor'
+            },
+            exec: function(widget, args, request) {
+                //row of the cursor on current line
+                var row = widget.getCursorPosition().row;
+                //last column of the cursor on current line
+                var lastCol = widget.getSession().getValue().length;
```

Comment:
should be good to go now

On Thu, Apr 23, 2015 at 2:51 PM, Gordon Woodhull <notifications@github.com>
wrote:

> In htdocs/js/ui_utils.js
> <https://github.com/att/rcloud/pull/1417#discussion_r28991808>:
>
> > +                //make sure it appears at beginning of text
> > +                widget.navigateLineStart();
> > +            }
> > +        } ,
> > +        {
> > +            name: 'execute-cursor-at-end',
> > +            bindKey: {
> > +
> > +                mac: 'Ctrl-E',
> > +                sender: 'editor'
> > +            },
> > +            exec: function(widget, args, request) {
> > +                //row of the cursor on current line
> > +                var row = widget.getCursorPosition().row;
> > +                //last column of the cursor on current line
> > +                var lastCol = widget.getSession().getValue().length;
>
> Here it is: document.getLine(n)
>
> Example:
> https://github.com/att/rcloud/blob/develop/htdocs/js/ui/command_prompt.js#L54-L57
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/att/rcloud/pull/1417/files#r28991808>.
>



-- 
Anatoliy